### ISSUE TITLE
implicit declaration of function error with elements functions when compiling to wasm

### ISSUE THREAD

**Issue by Sosthene00** [ElementsProject/libwally-core#256]
I'm trying to build libwally for wasm. I'm building it inside a container with this dockerfile:
```
FROM debian:buster

ARG LIBWALLY_CORE_VERSION

RUN apt-get update && apt-get install -yqq git \
    uncrustify \
    python3-distutils-extra \
    python3-dev \
    build-essential \
    libffi-dev \
    autoconf \
    libtool \
    pkg-config \
    lib32z1 \
    unzip \
    curl \
    apt-transport-https

RUN git clone https://github.com/emscripten-core/emsdk.git /src/emsdk

WORKDIR /src/emsdk
RUN ./emsdk install latest && ./emsdk activate latest

RUN git clone https://github.com/ElementsProject/libwally-core.git -b release_$LIBWALLY_CORE_VERSION /src/my_project/libwally
WORKDIR /src/my_project/libwally
RUN git submodule init && \
    git submodule sync --recursive && \
    git submodule update --init --recursive

ARG PYTHON_VERSION=3
ENV SOURCE_EMSDK='. /src/emsdk/emsdk_env.sh'
RUN ./tools/cleanup.sh && ./tools/autogen.sh
SHELL [ "/bin/bash", "-c" ]
ARG CFLAGS="-fno-stack-protector"
RUN ${SOURCE_EMSDK} && emconfigure ./configure \
    --build=$HOST_OS \
    ac_cv_c_bigendian=no \
    --disable-swig-python \
    --disable-swig-java \
    --enable-elements \
    --disable-ecmult-static-precomputation \
    --disable-tests \
    --enable-export-all 
RUN ${SOURCE_EMSDK} && emmake make -j$(nproc)
ADD wasm-module/ /src/my_project
WORKDIR /src/my_project
ARG EXTRA_EXPORTED_RUNTIME_METHODS="['getValue', 'UTF8ToString', 'stringToUTF8', 'lengthBytesUTF8', 'cwrap', 'ccall']"
ARG EXPORTED_FUNCTIONS="['_malloc','_free','_wally_init','_bip39_mnemonic_from_bytes','_wally_is_elements_build','wally_asset_blinding_key_from_seed']"
RUN ${SOURCE_EMSDK} && emcc \
    -s "EXTRA_EXPORTED_RUNTIME_METHODS=$EXTRA_EXPORTED_RUNTIME_METHODS" \
    -s "EXPORTED_FUNCTIONS=$EXPORTED_FUNCTIONS" \
    libwally/src/.libs/*.o \
    libwally/src/secp256k1/src/*.o \
    libwally/src/ccan/ccan/*/.libs/*.o \
    libwally/src/ccan/ccan/*/*/.libs/*.o \
    -o my_project.html \
    --shell-file contrib/shell_minimal.html
ENTRYPOINT [ "python3", "-m", "http.server" ]
```
Note that I pass the `--enable-elements` flag to `./configure`. 
But if I try to add an elements function to `EXPORTED_FUNCTIONS`, like `wally_asset_blinding_key_from_seed` for example, emcc will fail with the following error: `emcc: error: undefined exported symbol: "wally_asset_blinding_key_from_seed" [-Wundefined] [-Werror]`

I don't understand why elements function can't be exported, I used to build libwally with the python-swig and `--enable-elements` without problem, am I missing something ?

**Comment by Sosthene00** [comment]
Just noticed that I made a silly typo, I guess it means I could use some rest.

**Comment by Sosthene00** [comment]
...or maybe there's actually an issue with that.

I noticed that after compiling libwally to wasm I can call `wally_asset_blinding_key_from_seed` from js with `ccall`, but then I created a .c file that I compile with libwally to wasm and when I call the same function from there I have the following error on compilation:

`error: implicit declaration of function 'wally_asset_blinding_key_from_seed' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    if ((ret = wally_asset_blinding_key_from_seed(seed, sizeof(seed), bytes_out, sizeof(bytes_out))) != 0) {`

To clarify: 
* this works in a .js file: `ccall('wally_asset_blinding_key_from_seed', 'number', ['number', 'number', 'number', 'number'], [seed, seed.length, bytes_out, bytes_out.length]`
* but this doesn't work from a .c file: `wally_asset_blinding_key_from_seed(seed, sizeof(seed), bytes_out, sizeof(bytes_out))`

I called a lot of other libwally functions without problem, but I tried another relative to elements and I have the same error, so I guess that's something wrong with elements functions.
`wally_is_elements_build` returns 1.
